### DISCUSSION TITLE
3 sigma rule

### DISCUSSION THREAD

**Discussion by CheungBH** [open-mmlab/mmpose#545] [General] [answered]
Hi. I saw that when  generating a gaussian heatmap, I found that 3-sigma rule is followed, that the value of cut-off range is three times larger than the sigma value. I wonder whether there are any references about it? Whether it is useful to make any modifications on it to train a better model?

![Screenshot from 2021-03-30 11-04-11](https://user-images.githubusercontent.com/39186017/112927524-b62c6380-9147-11eb-841c-0d3b74363ebb.png)
Line 284

**Comment by jin-s13** [comment, marked as answer]
This is a common practice to generate the heatmap target, used by many recent papers (e.g. simple baseline, HRNet)
If dist > 3 sigma, the value is too small (near zero). In order to speed up the heatmap generation, we keep these values as zero.

**Comment by CheungBH** [comment]
Thanks for your explanation.

But I am still confused about the sigma and distance. I wonder whether the value of dist and sigma relevant to the output image size? As soon as I am concerned, when dist and sigma fixed, the size of the distribution will be fixed. However, as the output image size grows larger, the distribution will occupy a smaller area comparing with the whole image, meaning that it is easier for neural network to be make mistakes.
In this situation, how should we adjust the sigma and dist? I thought the sigma should be bigger as the image size grows larger, but I don't know whether I am right.

**Comment by jin-s13** [comment]
It sounds reasonable to also adjust the sigma/distance according to the input image size. But I am not sure what is the best choice. 
Currently, we **empirically** set sigma=2 for input size 256, and sigma=3 for input size 384. distance=3xsigma.

**Comment by CheungBH** [comment]
Thanks for your idea. Maybe the most common used one is the best one hhhhh.
